homework/12_B/13_Irina/Circles/src/Main.java

```diff
@@ -0,0 +1,46 @@
+import java.util.Objects;
+import java.util.Scanner;
+
+public class Main {
+    public static OverlayType crossType(Circle c1, Circle c2) {
+        if (c1.equals(c2)) {
+            return OverlayType.IDENTICAL;
+        }
+
+        double distanceBtwCenters = Math.sqrt(
+                Math.pow(c1.getCenter().getX() - c2.getCenter().getX(), 2) +
+                        Math.pow(c1.getCenter().getY() - c2.getCenter().getY(), 2)
+        );
+        //System.out.println("distanceBtwCenters =  " + distanceBtwCenters);
+
+        if (c1.getR() + c2.getR() == distanceBtwCenters) {
+            return OverlayType.TOUCHING;
+        }
+
+        if ( (c1.getR() > c2.getR() && (distanceBtwCenters + c2.getR() <= c1.getR())) ||
+                c2.getR() > c1.getR() && (distanceBtwCenters + c1.getR() <= c2.getR())) {
+            return OverlayType.ONE_INSIDE_OTHER;
+        }
+
+        if (c1.getR() + c2.getR() > distanceBtwCenters) {
```

Comment:
What if they are one inside the other but not touching.

Reply:
i don't understand what you mean. could you provide a graph?